packages/react-components/src/lib/context/AgoricProviderLite.tsx

```diff
@@ -0,0 +1,219 @@
+  agoricConverters,
+  makeAgoricWalletConnection,
+} from '@agoric/web-components';
+import {
+  type ChainStorageWatcher,
+  makeAgoricChainStorageWatcher,
+} from '@agoric/rpc';
+import { Registry } from '@cosmjs/proto-signing';
+import {
+  AminoTypes,
+  defaultRegistryTypes,
+  createBankAminoConverters,
+  createAuthzAminoConverters,
+} from '@cosmjs/stargate';
+import { subscribeLatest } from '@agoric/notifier';
+import type { ChainName } from 'cosmos-kit';
+import type { AssetKind } from '@agoric/ertp/src/types';
+
+type Props = {
+  chainName?: ChainName;
+  useCustomEndpoints?: boolean;
+  onConnectionError?: (e: unknown) => void;
+};
+
+/**
+ * Provides access to Agoric-specific account features such as smart wallet
+ * provisioning, purses, offer signing, and more.
+ *
+ * Meant to be used inside a `ChainProvider` from
+ * https://github.com/cosmology-tech/cosmos-kit/tree/main/packages/react.
+ *
+ * If a custom `ChainProvider` is not desired, one can implicitly use
+ * the default one via `AgoricProvider`.
+ */
+export const AgoricProviderLite = ({
+  children,
+  onConnectionError = () => { },
+  chainName = 'agoric',
+  useCustomEndpoints = true,
+}: PropsWithChildren<Props>) => {
+  const [walletConnection, setWalletConnection] = useState<
+    AgoricWalletConnection | undefined
+  >(undefined);
+  const [chainStorageWatcher, setChainStorageWatcher] = useState<
+    ChainStorageWatcher | undefined
+  >(undefined);
+  const [purses, setPurses] = useState<
+    PursesJSONState<AssetKind>[] | undefined
+  >(undefined);
+  const [offerIdsToPublicSubscribers, setOfferIdsToPublicSubscribers] =
+    useState<Record<string, Record<string, string>> | undefined>(undefined);
+  const [isSmartWalletProvisioned, setIsSmartWalletProvisioned] = useState<
+    boolean | undefined
+  >(undefined);
+
+  const { status, client } = useWalletClient();
+  const chain = useChain(chainName);
+
+  useEffect(() => {
+    let isCancelled = false;
+
+    const watchPurses = async () => {
+      assert(walletConnection);
+      const n = walletConnection.pursesNotifier;
+
+      for await (const result of subscribeLatest(n)) {
+        if (isCancelled) return;
+        if (!result) continue;
+        setPurses(result);
+      }
+    };
+
+    const watchSmartWalletProvision = async () => {
+      assert(walletConnection);
+      const n = walletConnection.smartWalletStatusNotifier;
+      for await (const status of subscribeLatest(n)) {
+        if (isCancelled) return;
+        if (!status) continue;
+        setIsSmartWalletProvisioned(status.provisioned);
+      }
+    };
+
+    const watchPublicSubscribers = async () => {
+      assert(walletConnection);
+      const n = walletConnection.publicSubscribersNotifier;
+
+      for await (const entries of subscribeLatest(n)) {
+        if (isCancelled) return;
+        if (!entries) continue;
+        const result = Object.fromEntries(entries);
+        setOfferIdsToPublicSubscribers(result);
```

Comment:
Do we want to expose a filter here? It is likely a consumer is concerned with a subset of publicSubscribers versus all of them

Reply:
Ok cool, agree there doesn't seem to be an elegant solution currently.

I think adding `useOffers(selectorFn)` or `useContinuingOffers(selectorFn)` with an example for vaults would be a nice addition.